Signal/src/Profiles/OWSProfileManager.m

```diff
@@ -398,6 +399,36 @@ - (void)writeAvatarToDisk:(UIImage *)avatar
     });
 }
 
+- (NSData *)processedImageDataForRawAvatar:(UIImage *)image
+{
+    NSUInteger kMaxAvatarBytes = 5 * 1000 * 1000;
+
+    if (image.size.width != kOWSProfileManager_MaxAvatarWidth
```

Comment:
I agree we should downsample oversize avatars, but do we want to upsample small avatars?  It seems like it can only cause harm.